openmls/src/tree/secret_tree.rs

```diff
@@ -102,6 +109,7 @@ impl SecretTree {
         });
 
         SecretTree {
+            own_index,
             nodes,
             handshake_sender_ratchets: vec![None; size.as_usize()],
```

Comment:
When we're rewriting this anyway we should try do something else here. Allocating two vectors of the size of the tree is pretty heavy and probably not really neded. Also, this requires you to allow cloning of keys, which is sort off the opposite of what you want to do with this PR.

Reply:
The rewrite was more focused on the `sender_ratchet` and the FS deficiency that results from keeping around past ratchet secrets rather than the aead keys and nonces. Here I'm just passing along an index. That being said, I enjoy this sort of work, so I'd be happy to rewrite the `secret_tree` as well, although it does distract somewhat from the 0.4 goal.

Reply:
Ok, I filed #789.